samples/extensions/mesh_shader_culling/README.md

```diff
@@ -0,0 +1,221 @@
+gl_MeshVerticesEXT[k * 4 + 0].gl_Position = vec4(2.0 * sharedData.subDimension * unitVertex_0, 0.0f, 1.0f) + sharedData.positionTransformation + displacement;
+gl_MeshVerticesEXT[k * 4 + 1].gl_Position = vec4(2.0 * sharedData.subDimension * unitVertex_1, 0.0f, 1.0f) + sharedData.positionTransformation + displacement;
+gl_MeshVerticesEXT[k * 4 + 2].gl_Position = vec4(2.0 * sharedData.subDimension * unitVertex_2, 0.0f, 1.0f) + sharedData.positionTransformation + displacement;
+gl_MeshVerticesEXT[k * 4 + 3].gl_Position = vec4(2.0 * sharedData.subDimension * unitVertex_3, 0.0f, 1.0f) + sharedData.positionTransformation + displacement;
+// Indices
+gl_PrimitiveTriangleIndicesEXT[k * 2 + 0] = unitPrimitive_0 + k * uvec3(4);
+gl_PrimitiveTriangleIndicesEXT[k * 2 + 1] = unitPrimitive_1 + k * uvec3(4);
+// Assigning the color output:
+vec3 color = vec3(1.0f, 0.0f, 0.0f) * (k + 1) / sharedData.meshletsNumber;
+outColor[k * 4 + 0] = color;
+outColor[k * 4 + 1] = color;
+outColor[k * 4 + 2] = color;
+outColor[k * 4 + 3] = color;
+```
+
+More details of meshlets generation can be found in the attached article:
+
+[Using Mesh Shaders for Professional Graphics](https://developer.nvidia.com/blog/using-mesh-shaders-for-professional-graphics/)
+
+## Meshlets culling
+
+This sample uses a simple cull functionality from the mesh 
+shader. In mesh shading, there is no command, or 
+automated functions that helps to "cull" a geometry. The cull logic simply determines the condition,
+whether a meshlet may be generated or not. 
+
+In this sample, a circular visual zone is centered at the origin, with an adjustable radius, controlled by the gui.
+Whenever a meshlet moves out of the visual zone, its generation process will be skipped.
+
+```glsl
+// the actual position of each meshlet:
+vec4 position = displacement + sharedData.positionTransformation;
+float squareRadius = position.x * position.x + position.y * position.y;
+// Cull Logic: only is the meshlet center position is within the view circle defined by the cull radius,
+// then this very meshlet will be generated. Otherwise, meshlet will NOT be generated.
+if (squareRadius < sharedData.cullRadius * sharedData.cullRadius)
+{
+    // Generating meshlets
+}
+```
+
+There are more adaptive theories for such purpose, and a 
+commonly used cull logic can be found in the
+following video:
+
+[Culling with NVIDIA Mesh Shaders](https://www.youtube.com/watch?v=n3cnUHYGbpw)
+
+## UI overlay
+
+![Sample](./images/mesh_shader_culling_ui_overlay.png)
+
+A simple UI overlay is created to help showcase the mesh shader cull feature. Where, the radius of the circular cull
+mask can be adjusted using the "Cull Radius" slider, which ranges from ```0.5f``` to ```2.0f```. And the meshlet density
+level can be controlled using the combo-box. Where, a selection of 4-by-4, 6-by-6, and 8-by-8 of density opinions is
+provided. Where, users may use keyboard input ```W``` ```A``` ```S``` ```D``` to move the square mesh around the scene.
```

Comment:
Maybe it's because I'm not a native speaker, but those "Where," make this cumbersome to read.